app/controllers/api/host_aggregates_controller.rb

```diff
@@ -25,6 +25,32 @@ def delete_resource_main_action(_type, host_aggregate, _data = {})
       {:task_id => host_aggregate.delete_aggregate_queue(current_user.userid)}
     end
 
+    def add_host_resource(type, id, data = {})
+      raise if id.nil? || data["host_id"].nil?
+
+      host_aggregate = resource_search(id, type)
+      raise unless host_aggregate.supports?(:add_host)
```

Comment:
Should return the unsupported reason,
```suggestion
      raise BadRequestError, host_aggregate.unsupported_reason(:add_host) unless host_aggregate.supports?(:add_host)
```